platoon/mpi_util.py

```diff
@@ -34,9 +34,9 @@ def launch_mpi_workers(workers_count, experiment_name, worker_args):
     args = shape_args(experiment_name, worker_args, "worker")
     info = MPI.Info.Create()
     info['host'] = socket.gethostname()
-    #  info['ompi_non_mpi'] = 'true'
     env = dict(os.environ)
-    info['env'] = '\n'.join(env.keys())
+    keep_list = ['THEANO_FLAGS', 'PLATOON_TEST_WORKERS_NUM', '_']
+    info.Set(key='env', value='\n'.join('%s=%s' % (key, env[key]) for key in keep_list if key in env))
```

Comment:
Are you sure the `<key>=<value>` is correct? Has this been able to pass environmental variables correctly to children?

Also, if we were to be concrete about which variables we would like to pass, we should be ignorant about any variable name except `THEANO_FLAGS`. This means that `_` and `PLATOON_TEST_WORKERS_NUM` should be explicitly passed. BTW, what does env var `_` correspond to?

We can also try without passing anything to the info object and see what environmental variables has a worker process on its "own". For example, try commenting out `info.Set(...)` on this file and place a `print(dict(os.environ))` before and after [line 301](https://github.com/tsirif/platoon/blob/fix_attempt/platoon/channel/worker.py#L301) in Worker, so that we see what is the default for openMPI implementation in dynamically process spawning situations (maybe it is to pass all parent's environmental variables...).